packages/rollup-full-node/src/app/handler/routing-handler.ts

```diff
@@ -0,0 +1,171 @@
+   */
+  public async handleRequest(
+    method: string,
+    params: any[],
+    sourceIpAddress: string
+  ): Promise<string> {
+    log.debug(
+      `Proxying request for method [${method}], params: [${JSON.stringify(
+        params
+      )}]`
+    )
+    let tx: Transaction
+
+    if (
+      method === Web3RpcMethods.sendTransaction ||
+      method === Web3RpcMethods.sendRawTransaction
+    ) {
+      try {
+        tx = parseTransaction(params[0])
+      } catch (e) {
+        // means improper format -- since we can't get address, add to quota by IP
+        this.validateRateLimit(undefined, sourceIpAddress)
+        throw new InvalidParametersError()
+      }
+
+      this.validateRateLimit(tx.from)
+    } else {
+      this.validateRateLimit(undefined, sourceIpAddress)
+    }
+
+    this.assertDestinationValid(tx)
+
+    try {
+      const result: any =
+        methodsToRouteWithTransactionHandler.indexOf(method) >= 0
+          ? await this.transactionClient.handle<string>(method, params)
+          : await this.readOnlyClient.handle<string>(method, params)
+      log.debug(
+        `Request for [${method}], params: [${JSON.stringify(
+          params
+        )}] got result [${result}]`
+      )
+      return result
+    } catch (e) {
+      logError(
+        log,
+        `Error proxying request: [${method}], params: [${JSON.stringify(
+          params
+        )}]`,
+        e
+      )
+      throw e
+    }
+  }
+
+  /**
+   * Validates the request is under the rate limit for the provided tx from address or
+   * source IP address if a rate limit is configured
+   *
+   * @param txFromAddress The from address only provided if this is to check against the tx rate limit.
+   * @param sourceIpAddress The IP address only provided if this is to check against the IP rate limit.
+   * @throws AccountRateLimiter If the IP in question is above its rate limit.
+   * @throws TransactionLimitError If the address in question is above its rate limit.
+   */
+  private validateRateLimit(
+    txFromAddress?: string,
+    sourceIpAddress?: string
+  ): void {
+    if (!this.accountRateLimiter) {
+      return
+    }
+    if (!!txFromAddress) {
+      this.accountRateLimiter.validateTransactionRateLimit(txFromAddress)
+    } else {
+      this.accountRateLimiter.validateRateLimit(sourceIpAddress)
+    }
+  }
+
+  /**
+   * If provided a transaction, and a transaction destination whitelist is configured,
+   * this will make sure the destination of the transaction is on the whitelist or
+   * the transaction is sent by the deployer address.
+   *
+   * @param tx The transaction in question.
+   * @throws InvalidTransactionDesinationError if the transaction destination is invalid.
+   */
+  private assertDestinationValid(tx?: Transaction): void {
+    if (
+      !!tx &&
+      !!this.toAddressWhitelist.length &&
+      !(tx.to in this.toAddressWhitelist) &&
+      tx.from !== this.deployAddress
```

Comment:
@karlfloersch and I recently discussed that it would be a pretty awesome to be able to limit the destination to certain methods as well as addresses.  In practice I guess that would look like adding a `methodIdWhitelist` for each `toAddressWhitelist` and comparing first 4 bytes of `tx.data` here.

I don't think it's critical--but if not too hard to add, could be a valuable extra level of defense 😀

Reply:
Added an RPC method whitelist, but not a tx data methodId whitelist. Would take a bit of time to configure, so leaving that as a nice-to-have if we get to it.